### ISSUE TITLE
Long Pause on Gen1 GC

### ISSUE THREAD

**Issue by dickens-code** [dotnet/runtime#88426]
### Description

Recently my team has been working on lowering down memory allocation to minimize GC pressure.
While we used different technique to reuse/recycle object, or use ArrayPool.Shared to rent/return memory, we found that there long periodic long pauses (~1s) of Gen0/Gen1 GC few times per hour.
We used PerfView command to spot it out:
```
PerfView -NoGui -AcceptEULA -NoNGenRundown -BufferSizeMB:2048 -CircularMB:2048 -Zip:false -Merge:true -ClrEvents:GC+Stack -StopOnGCOverMSec:300 -KernelEvents:Memory,VirtualAlloc,Default -OnlyProviders:ClrPrivate:1:5,Clr:0x40000001:5 -NoRundown
-DelayAfterTriggerSec:0 -CollectMultiple:1 -DataFile:XXX-2307051821-48272-gcsg -FocusProcess:48272 -Process:48272 collect
```

GC Stats showing 1s Gen1 GC pause
![Screenshot 2023-07-05 225019](https://github.com/dotnet/runtime/assets/23349139/3423eaed-d7b7-43ca-9a24-bbfca7463d4a)
[pic1]

When looking into the CPU Stack of profile, we discovered that these few calls during GC causes most part of delay:
1. coreclr!SyncBlockCache::InsertCleanupSyncBlock
2. coreclr!SVR::GCHeap::IsPromoted
3. ntoskrnl!MiRemoveLowestPriorityStandbyPage
![Screenshot 2023-07-05 224639](https://github.com/dotnet/runtime/assets/23349139/d032ba0b-f51c-4562-8917-b82275eb0bc0)
[pic2]

When Goto Source these lines are most time consuming ...
InsertCleanupSyncBlock
![Screenshot 2023-07-05 231740](https://github.com/dotnet/runtime/assets/23349139/9de7cbc8-eddc-4ad7-b437-47eb0491cf2d)
[pic3]

IsPromoted
![Screenshot 2023-07-05 231840](https://github.com/dotnet/runtime/assets/23349139/4dead6e8-5ff6-4997-b1ed-7d97e431208d)
[pic4]

### Configuration
.NET core 7
Windows Server 2012 x64

**Comment by mangod9** [comment]
Hello @dickens-code , would it possible to securely share the traces you have collected? Also are you observing that this has regressed in .NET 7?

**Comment by dickens-code** [comment]
yes, it is happening to net7, however I can't share the profile (etl file) to you due to security reason. Anything I should check?

**Comment by mangod9** [comment]
would it be possible to collect some GCCollectOnly traces? That shouldnt have much PII or other info.

**Comment by cshung** [comment]
@mangod9, I am working with @dickens-code offline to diagnose the issue. We will report findings here whenever we make progress.

**Comment by dickens-code** [comment]
thanks both @mangod9 and @cshung ~
